packages/serializer/src/json.js

```diff
@@ -1476,19 +1476,25 @@ const JSONSerializer = Serializer.extend({
       const extracted = {};
 
       payload.errors.forEach((error) => {
-        if (error.source && error.source.pointer) {
-          let key = error.source.pointer.match(SOURCE_POINTER_REGEXP);
-
-          if (key) {
-            key = key[2];
-          } else if (error.source.pointer.search(SOURCE_POINTER_PRIMARY_REGEXP) !== -1) {
-            key = PRIMARY_ATTRIBUTE_KEY;
-          }
-
-          if (key) {
-            extracted[key] = extracted[key] || [];
-            extracted[key].push(error.detail || error.title);
-          }
+        const errorPointer = error.source?.pointer ?? '/data'
```

Comment:
see: https://jsonapi.org/format/#error-objects
allows for `source` to be **omitted**
> source: an object containing references to the primary source of the error. It SHOULD include one of the following members or be omitted: